src/libcommonserver/io/iohelper.cpp

```diff
@@ -323,6 +323,67 @@ bool IoHelper::getFileSize(const SyncPath &path, uint64_t &size, IoError &ioErro
     return true;
 }
 
+bool IoHelper::getDirectorySize(const SyncPath &path, uint64_t &size, unsigned int maxDepth, bool &skipedTooDeep,
+                                IoError &ioError) {
+    ItemType itemType;
+    const bool success = getItemType(path, itemType);
+    ioError = itemType.ioError;
+    if (!success || ioError != IoErrorSuccess) {
+        return _isExpectedError(ioError);
+    }
+
+    assert(ioError != IoErrorUnknown);
+
+    if (itemType.nodeType != NodeTypeDirectory) {
+        ioError = IoErrorIsAFile;
+        return false;
+    }
+
+    IoHelper::DirectoryIterator dir;
+    IoHelper::getDirectoryIterator(path, true, ioError, dir);
+    if (ioError != IoErrorSuccess) {
+        LOG_WARN(Log::instance()->getLogger(), "Error in DirectoryIterator: " << Utility::formatIoError(path, ioError).c_str());
+        return _isExpectedError(ioError);
+    }
+
+    DirectoryEntry entry;
+    ioError = IoErrorSuccess;
+    bool endOfDirectory = false;
+    size = 0;
+    while (dir.next(entry, endOfDirectory, ioError) && !endOfDirectory) {
+        if (entry.is_directory()) {
+            if (maxDepth == 0) {
+                LOG_WARN(Log::instance()->getLogger(), "Max depth reached in getDirectorySize, skipping deeper directories: "
+                                                           << Utility::formatSyncPath(path).c_str());
+                skipedTooDeep = true;
+                continue;
+            }
+            uint64_t entrySize;
+            if (!getDirectorySize(entry.path(), entrySize, maxDepth - 1, skipedTooDeep, ioError)) {
+                return false;
+            }
+            size += entrySize;
+        }
+        uint64_t entrySize;
```

Comment:
Define it on line 369 while initialiazing it

Reply:
Done in commits: https://github.com/Infomaniak/desktop-kDrive/pull/100/commits/13ac91508fc1f358d2f9436294069c2f74ce17c0  & (https://github.com/Infomaniak/desktop-kDrive/pull/100/commits/0ec1337d65aca128d7f3f0ac74bbabb40b9e8ef7